File: app/src/main/java/com/mmutert/freshfreezer/data/converters/ConditionConverter.java
package com.mmutert.freshfreezer.data.converters;

import androidx.room.TypeConverter;

import com.mmutert.freshfreezer.data.Condition;


public class ConditionConverter {

    @TypeConverter
    public static String fromCondition(Condition unit) {

        return unit.toString();
    }


    @TypeConverter
    public static Condition toCondition(String unit) {

        return Condition.valueOf(unit);
    }
}

File: app/src/main/java/com/mmutert/freshfreezer/data/converters/ConditionConverter.kt
package com.mmutert.freshfreezer.data.converters

import androidx.room.TypeConverter
import com.mmutert.freshfreezer.data.Condition

object ConditionConverter {
    @JvmStatic
    @TypeConverter
    fun fromCondition(unit: Condition): String {
        return unit.toString()
    }

    @JvmStatic
    @TypeConverter
    fun toCondition(unit: String?): Condition {
        return Condition.valueOf(unit!!)
    }
}